service/src/main/java/bio/terra/workspace/db/WorkspaceDao.java

```diff
@@ -222,145 +230,131 @@ private boolean updateWorkspaceColumns(UUID workspaceId, MapSqlParameterSource c
    */
-  @WriteTransaction
-  public void deleteGcpCloudContext(UUID workspaceId) {
-    deleteGcpCloudContextWorker(workspaceId);
+  private Optional<String> getCloudContextWorker(UUID workspaceId, CloudPlatform cloudPlatform) {
+    String sql =
+        "SELECT context FROM cloud_context "
+            + "WHERE workspace_id = :workspace_id AND cloud_platform = :cloud_platform";
+    MapSqlParameterSource params =
+        new MapSqlParameterSource()
+            .addValue("workspace_id", workspaceId.toString())
+            .addValue("cloud_platform", cloudPlatform.toSql());
+    try {
+      return Optional.ofNullable(
+          DataAccessUtils.singleResult(
+              jdbcTemplate.query(sql, params, (rs, rowNum) -> rs.getString("context"))));
+    } catch (EmptyResultDataAccessException e) {
+      return Optional.empty();
+    }
   }
 
   /**
-   * Delete a cloud context for the workspace validating the projectId
+   * Common worker for deleting cloud context
    *
-   * @param workspaceId workspace of the cloud context
-   * @param projectId the GCP project id to validate
+   * @param workspaceId workspace holding the context
+   * @param cloudPlatform platform of the context
+   * @param flightId if non-null, then it is compared to the creating_flight to make sure a
+   *     conflicting create does not delete a valid cloud context
    */
-  @WriteTransaction
-  public void deleteGcpCloudContextWithIdCheck(UUID workspaceId, String projectId) {
-    // Only perform the delete, if the project id matches the input project id
-    Optional<GcpCloudContext> gcpCloudContext = getGcpCloudContext(workspaceId);
-    if (gcpCloudContext.isPresent()) {
-      if (StringUtils.equals(projectId, gcpCloudContext.get().getGcpProjectId())) {
-        deleteGcpCloudContextWorker(workspaceId);
-      }
-    }
-  }
-
-  private void deleteGcpCloudContextWorker(UUID workspaceId) {
-    final String sql =
+  private void deleteCloudContextWorker(
+      UUID workspaceId, CloudPlatform cloudPlatform, @Nullable String flightId) {
+    final String platform = cloudPlatform.toSql();
+    String sql =
         "DELETE FROM cloud_context "
             + "WHERE workspace_id = :workspace_id AND cloud_platform = :cloud_platform";
 
     MapSqlParameterSource params =
         new MapSqlParameterSource()
             .addValue("workspace_id", workspaceId.toString())
-            .addValue("cloud_platform", CloudPlatform.GCP.toString());
+            .addValue("cloud_platform", platform);
+
+    if (StringUtils.isNotEmpty(flightId)) {
+      sql = sql + " AND creating_flight = :flightid";
+      params.addValue("flightid", flightId);
+    }
 
     int rowsAffected = jdbcTemplate.update(sql, params);
     boolean deleted = rowsAffected > 0;
 
     if (deleted) {
-      logger.info("Deleted GCP cloud context for workspace {}", workspaceId);
+      logger.info("Deleted {} cloud context for workspace {}", platform, workspaceId);
```

Comment:
should these and similar loggings be `DEBUG`?

Reply:
We are consistently logging all DAO operations as info. I don't think it is too noisy. Do you?

Reply:
No, that's fine. It's rare for us to call lots of these methods in rapid succession.